client/src/components/Layout/index.tsx

```diff
@@ -40,12 +71,29 @@ export function Header() {
         </Link>
 
         <NavigationBar />
-        <a
-          className="text-white hover:cursor-pointer hover:underline"
-          href="/api/logout"
-        >
-          Logout
-        </a>
+
+        {/* User dropdown */}
+        <div className="relative" ref={menuRef}>
+          <button
+            type="button"
+            onClick={() => setMenuOpen((prev) => !prev)}
+            className="font-public-sans hover:bg-blue-cool-70 flex items-center gap-2 rounded px-3 py-2 text-white focus:outline-none"
+          >
```

Comment:
There is currently no way for a screen reader to know this button opens up a menu. We should probably add a few aria attributes here. I _think_ something like this would work.
```suggestion
          <button
            type="button"
            aria-haspopup="menu"
            aria-expanded={menuOpen}
            onClick={() => setMenuOpen((prev) => !prev)}
            className="font-public-sans hover:bg-blue-cool-70 flex items-center gap-2 rounded px-3 py-2 text-white focus:outline-none"
          >
```

Reply:
Disregard this if we go with my other suggestion to use the headless UI component.